app/src/main/java/com/kickstarter/viewmodels/PledgeFragmentViewModel.kt

```diff
@@ -215,20 +216,25 @@ interface PledgeFragmentViewModel {
                     .compose(bindToLifecycle())
                     .subscribe(this.shippingRulesAndProject)
 
-            rulesAndProject
-                    .compose(bindToLifecycle())
+            val hasNoShippingRules = rulesAndProject
                     .map { ObjectUtils.isNull(it.first) || it.first.isEmpty() }
+
+            val isNoReward = reward
+                    .filter { RewardUtils.isNoReward(it) }
```

Comment:
You can just do 
`.map { RewardUtils.isNoReward(it) }`

Reply:
I added the `.filter { RewardUtils.isNoReward(it) }` here because we only want this to emit if it's a  `NoReward`. Without the filter, the `this.shippingRulesSectionIsGone` and `this.estimatedDeliveryInfoIsGone` will emit twice if the rewards `isReward`.
Here I'm trying to avoid a situation where `this.shippingRulesSectionIsGone` would emit `false` (because `RewardUtils.isNoReward(it)` would be false) and then emit `true` if the reward doesn't have shipping  rules.

Reply:
but won't it emit the same values? then you can do `distinctUntilChange`

Reply:
Not on the situation I mentioned above. It would emit `false` and then `true`, so the `distinctUntilChange` would not work.

Reply:
I edited my comment to try to make it more clear. So in the situation I mentioned above, the `disctinctUntilChange` will not work, since the values are different.